ecl/hql/hqlgram.y

```diff
@@ -8812,6 +8812,10 @@ simpleDataSet
 
                             bool hasFileposition = getBoolAttributeInList(extra, filepositionAtom, true);
                             record.setown(parser->checkIndexRecord(record, $5, extra));
+
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
+                                extra.setown(createComma(extra.getClear(), createAttribute(_disallowed_Atom)));
```

Comment:
1) The check+error would be in 1 place.
2) Avoid cost of checking access for all file constructs.

Probably the cost is inconsequential even if there are a large amount file/attribute checks of checks that don't end up being used.

My guess was there was a good reason why it's easier to check every attribute and flag at parse time, rather than do when attribute is actually used..